opensearch/src/main/java/org/opensearch/sql/opensearch/storage/script/filter/lucene/LuceneQuery.java

```diff
@@ -227,14 +238,31 @@ private ExprValue cast(FunctionExpression castFunction) {
           .put(
               BuiltinFunctionName.CAST_TO_TIMESTAMP.getName(),
               expr -> {
-                if (expr.type().equals(ExprCoreType.STRING)) {
+                ZonedDateTime zonedDateTime = getParsedDateTime(expr);
+                if (zonedDateTime != null) {
+                  return new ExprTimestampValue(zonedDateTime.toInstant());
+                } else if (expr.type().equals(ExprCoreType.STRING)) {
                   return new ExprTimestampValue(expr.valueOf().stringValue());
                 } else {
                   return new ExprTimestampValue(expr.valueOf().timestampValue());
                 }
               })
           .build();
 
+  /**
+   * Parses the date/time from the given expression if it is of string type and the reference type
+   * is an instance of OpenSearchDateType.
+   *
+   * @param expr The expression to parse.
+   * @return The parsed ZonedDateTime or null if the conditions are not met.
+   */
+  private ZonedDateTime getParsedDateTime(LiteralExpression expr) {
+    if (expr.type().equals(ExprCoreType.STRING) && this.ref.type() instanceof OpenSearchDateType) {
```

Comment:
line 220 check the type is STRING. can we simiply the code?

Reply:
can u update on this behaviour. https://github.com/opensearch-project/sql/blob/main/docs/user/general/datatypes.rst#conversion-from-string-to-date-and-time-types

we should clarify in doc what is the valide dateformat can be used for literal value. 
does it means only when the operator include indexDateField and literal value, then the literal value can be indexDateFieldd format?

Reply:
Added the doc